subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/JavaToolchainRepository.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.services.BuildService;
+import org.gradle.api.services.BuildServiceParameters;
+
+import java.net.URI;
+import java.util.Optional;
+
+/**
+ * Interface that needs to be implemented by Toolchain SPI plugins, in order to
+ * extend Gradle with the spec-to-URI logic required by Java toolchain auto-provisioning.
+ *
+ * @since 7.6
+ */
+@Incubating
+public interface JavaToolchainRepository extends BuildService<BuildServiceParameters.None> {
```

Comment:
This should be renamed to something like `JavaToolchainResolver` to match the DSL

Reply:
I would rather rename the DSL to match this. To me "resolver" is just another vague term without much meaning, while "repository" seems the better fit. But it's a subjective opinion, let's discuss it.